google/web-stories-wp
# Design panel: add context menu to layer panel

## Context

Add our fancy dandy context menu to the layer panel

## Summary

Adds the right click menu to the layer panel. Right clicking on a layer will open the context menu in the same way as right clicking on an element in the canvas.

**NOTE**: Right clicking a layer and the right clicking somewhere else re-opens the _same_ right click menu, since the selected element won't change. This is very apparent in the layer panel. If a user wants to right click a different layer while the context menu is open then it'll re-open the same menu where the user clicked.

This is the same functionality as the right click menu in the canvas.

Clicking an action in the menu will perform that action on the selected element.

## Relevant Technical Choices

Selecting multiple layers and `shift + right click` would still open the context menu. Added some code to stop that nonsense.

## To-do

n/a

## User-facing changes

<img src="https://user-images.githubusercontent.com/22185279/137528035-dba592ea-342e-4120-bc54-bd00627b1fb5.gif" height="400px" />

## Testing Instructions





This PR can be tested by following these steps:

1. Right clicking a layer should open the context menu for that element type
2. All right click actions should work as intended
3. Selecting multiple layers while holding `Shift` and then right clicking one while holding `Shift` should not open our cusom context menu 

## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #9139
